### DISCUSSION TITLE
Trouble Identifying Server Origin in MessageReceived Event

### DISCUSSION THREAD

**Discussion by Motzi10** [jchristn/WatsonWebsocket#140] [General]
Hi there!
I've developed a GUI that allows me to create multiple Clients connecting to different ports on the Server. However, I'm encountering an issue: I can't determine, within the MessageReceived Event on the client, which specific Server sent the message. In SuperSimpleTcp, there was an 'e.IpPort' option for this purpose, but it doesn't seem to be available here. Thanks in advance for any guidance or suggestions you can offer!

**Comment by jchristn** [comment]
Correct, the assumption is that the client knows to which server they've connected.  I'm assuming you're sharing the same MessageReceived event across multiple server instances.

**Reply by Motzi10** [reply]
Yes, I do!
So there is no way to determine which client triggered the MessageReceived event, if all clients are += attached to the same MessageReceived when created?
